packages/backend/src/open_payments/payment_pointer/middleware.ts

```diff
@@ -14,7 +14,8 @@ export function createPaymentPointerMiddleware() {
       const paymentPointer = await paymentPointerService.getByUrl(
         ctx.paymentPointerUrl
       )
-      if (!paymentPointer) {
+
+      if (!paymentPointer || !paymentPointer.isActive()) {
```

Comment:
Wasn't entirely sure if throwing the 404 was the right behavior (or best error code). If it is, I also considered not adding this condition and instead changing the query to get the payment pointer to filter out inactive ones (either new arg on `getByUrl` or new method like `getActiveByUrl`). Seems like fetching the pointer would be unnecessary unless we were actually going to handle it differently than an unfound one.

Reply:
I think that is the correct behavior. The middleware is used when fetching payment pointer details prior to another action. If the payment pointer is inactive, the other action should not be possible. But @mkurapov correct me if I'm wrong.

Reply:
This makes behaviour is what I was thinking of 👍

Reply:
```suggestion
      if (!paymentPointer?.isActive()) {
```
some sugar if you'd like